spec/support/stub_api_requests.rb

```diff
@@ -0,0 +1,72 @@
+module StubApiRequests
+
+  def stub_api_get(service_path, stub_fixture="success.json", opts={}, &block)
+    stub_api_request(:get, service_path, stub_fixture, nil, opts, &block)
+  end
+
+  def stub_api_delete(service_path, stub_fixture="success.json", opts={}, &block)
+    stub_api_request(:delete, service_path, stub_fixture, nil, opts, &block)
+  end
+
+  def stub_api_post(service_path, body, stub_fixture="success.json", opts={}, &block)
+    stub_api_request(:post, service_path, stub_fixture, body, opts, &block)
+  end
+
+  def stub_api_put(service_path, body, stub_fixture="success.json", opts={}, &block)  
+    stub_api_request(:put, service_path, stub_fixture, body, opts, &block)
+  end
+
+  def expect_api_request(meth, service_path, body=nil, opts={})
+    args = with_args(service_path, opts, body)
+    expect(a_request(meth, full_path(service_path)).with(args))
+  end
```

Comment:
This is the only new method. All of the rest of the refactoring was to support this, and because I couldn't help it. 😞